Split font related css property parsing out into its own file

weinig opened the pull request:
#### ffc9f55a00906bbbded0de84eb0fff957d073537
<pre>
Split font related css property parsing out into its own file
<a href="https://bugs.webkit.org/show_bug.cgi?id=278034">https://bugs.webkit.org/show_bug.cgi?id=278034</a>

Reviewed by Darin Adler.

Split font related CSS property parsing into its own file, consolidating
parts from CSSPropertyParserHelpers.h/cpp and CSSPropertyParserWorkerSafe.h/cpp.

Do a minor cleanup of names, but otherwise, the code has been copied as is.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFontFaceSet.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/FontFace.cpp:
* Source/WebCore/css/ShorthandSerializer.cpp:
* Source/WebCore/css/StyleProperties.cpp:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.h:
* Source/WebCore/css/parser/CSSSupportsParser.cpp:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/editing/EditingStyle.cpp:
* Source/WebCore/html/canvas/CanvasRenderingContext2D.cpp:
* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
* Source/WebCore/html/canvas/OffscreenCanvasRenderingContext2D.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleResolveForFontRaw.cpp:

Canonical link: <a href="https://commits.webkit.org/282527@main">https://commits.webkit.org/282527@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ce2de7d6d80ce1c7bfce7ecebe1dc02ab9db67bb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63410 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42766 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16007 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67431 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14018 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65530 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50454 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14298 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/51068 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/9693 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66479 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/39707 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54914 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/31752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/36385 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12270 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12890 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/57919 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12595 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69127 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7357 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/12185 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/58377 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7389 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54989 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/58616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14044 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/88/builds/6142 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/38587 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/39666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/40778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/39409 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2024 Samuel Weinig <sam@webkit.org>
```
If this is moved code, then does there need to be some other copyright too?

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
```diff
@@ -0,0 +1,219 @@
+
+#pragma once
+
+#include "CSSPropertyParserConsumer+RawTypes.h"
+#include "CSSValueKeywords.h"
+#include "SystemFontDatabase.h"
+#include <optional>
+#include <wtf/RefPtr.h>
+#include <wtf/Vector.h>
+#include <wtf/text/AtomString.h>
```
Is this the minimal set of includes? I trust that it is.

weinig committed:
Split font related css property parsing out into its own file
https://bugs.webkit.org/show_bug.cgi?id=278034

Reviewed by Darin Adler.

Split font related CSS property parsing into its own file, consolidating
parts from CSSPropertyParserHelpers.h/cpp and CSSPropertyParserWorkerSafe.h/cpp.

Do a minor cleanup of names, but otherwise, the code has been copied as is.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFontFaceSet.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/FontFace.cpp:
* Source/WebCore/css/ShorthandSerializer.cpp:
* Source/WebCore/css/StyleProperties.cpp:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.h:
* Source/WebCore/css/parser/CSSSupportsParser.cpp:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/editing/EditingStyle.cpp:
* Source/WebCore/html/canvas/CanvasRenderingContext2D.cpp:
* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
* Source/WebCore/html/canvas/OffscreenCanvasRenderingContext2D.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleResolveForFontRaw.cpp:

Canonical link: https://commits.webkit.org/282527@main

rkirsling commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
```diff
@@ -0,0 +1,1344 @@
+
+#else
+
+RefPtr<CSSPrimitiveValue> consumeFontStretch(CSSParserTokenRange& range)
```
Hmm, PlayStation is seeing the following here:
```
In file included from C:/BW/PlayStation-Debug-Build/build/WebKitBuild/Debug/WebCore/DerivedSources/unified-sources/UnifiedSource-2f84417a-23.cpp:1:
In file included from C:/BW/PlayStation-Debug-Build/build/Source/WebCore\css/parser/CSSPropertyParserConsumer+Angle.cpp:26:
In file included from C:/BW/PlayStation-Debug-Build/build/Source/WebCore\css/parser\CSSPropertyParserConsumer+Angle.h:27:
In file included from C:/BW/PlayStation-Debug-Build/build/Source/WebCore\css/parser\CSSParserToken.h:32:
In file included from C:/BW/PlayStation-Debug-Build/build/Source/WebCore/css\CSSPrimitiveValue.h:24:
In file included from C:/BW/PlayStation-Debug-Build/build/Source/WebCore/css\CSSAnchorValue.h:28:
In file included from C:/BW/PlayStation-Debug-Build/build/Source/WebCore/css\CSSValue.h:25:
C:/BW/PlayStation-Debug-Build/build/WebKitBuild/Debug/WTF/Headers\wtf/RefPtr.h:53:80: error: cannot initialize a member subobject of type 'typename PtrTraits::StorageType' (aka 'WebCore::CSSPrimitiveValue *') with an rvalue of type 'WebCore::CSSValue *'
   53 |     template<typename X, typename Y, typename Z> RefPtr(RefPtr<X, Y, Z>&& o) : m_ptr(o.leakRef()) { }
      |                                                                                ^     ~~~~~~~~~~~
C:/BW/PlayStation-Debug-Build/build/Source/WebCore\css/parser/CSSPropertyParserConsumer+Font.cpp:763:16: note: in instantiation of function template specialization 'WTF::RefPtr<WebCore::CSSPrimitiveValue>::RefPtr<WebCore::CSSValue, WTF::RawPtrTraits<WebCore::CSSValue>, WTF::DefaultRefDerefTraits<WebCore::CSSValue>>' requested here
  763 |         return result;
      |                ^
```

Having a surprising amount of trouble determining the missing include, but based on how this is used, the return value doesn't shouldn't need to be as specific as `RefPtr<CSSPrimitiveValue>` anyway...?
